app/models/favorite.rb

```diff
@@ -0,0 +1,7 @@
+class Favorite < ActiveRecord::Base
+  belongs_to :tweet
+  belongs_to :user
+
+  validates :user_id, presence: true
+  validates :tweet_id, presence: true
```

Comment:
So, you should be able to rewrite this as `validates :user, presence: true`. Very nitpicky. But that's how we started writing our validations in models like this because it tends to be a little more readable. If you try this and it doesn't work, let me know. There may be another step I don't remember.